bin/install_swiftpm_dependencies

```diff
@@ -1,30 +1,91 @@
 #!/bin/bash -eu
 
-CUSTOM_PACKAGE_RESOLVED_PATH=${1-}
-BUILD_TYPE=${2-}
+print_usage() {
+  echo "Usage:"
+  echo "  Using the \`Package.resolved\` of an Xcode \`.xworkspace\`:"
+  echo "    $0 --workspace PATH --scheme NAME"
+  echo "  Using the \`Package.resolved\` of an Xcode \`.xcodeproj\`:"
+  echo "    $0 --project PATH"
+  echo "  Using the \`Package.resolved\` at the root, managed by \`swift package\` (and not Xcode)"
+  echo "    $0 --use-spm"
+}
+
+# Parse input parameters
+XCWORKSPACE_PATH=
+SCHEME_NAME=
+XCODEPROJ_PATH=
+USE_SPM=false
+while [[ "$#" -gt 0 ]]; do
+	case $1 in
+		--workspace)
+			XCWORKSPACE_PATH="$2"
+      shift
+			;;
+		--scheme)
+			SCHEME_NAME=$2
+      shift
+			;;
+    --project)
+      XCODEPROJ_PATH=$2
+      shift
+      ;;
+    --use-spm)
+      USE_SPM=true
+      ;;
+    -h|--help)
+      print_usage
+      exit 0
+      ;;
+    *)
+      echo "Unknown option or unexpected argument: $1"
+      print_usage
+      exit 2
+      ;;
+	esac
+	shift
+done
+
 SPM_CACHE_LOCATION="$HOME/Library/Caches/org.swift.swiftpm"
 
-WORKSPACE_PACKAGE_RESOLVED_PATH=(*.xcworkspace/xcshareddata/swiftpm/Package.resolved)
-ROOT_PACKAGE_RESOLVED_PATH=Package.resolved
+# Try to guess what to do if no option provided explicitly
+if [[ -z "${XCWORKSPACE_PATH}" && -z "${XCODEPROJ_PATH}" && "${USE_SPM}" != "true" ]]; then
+  echo "No \`--workspace\`, \`--project\` or \`--use-spm\` flag provided. Trying to guess the correct one to use..."
+  FOUND_ROOT_WORKSPACE_PATHS=(*.xcworkspace)
+  if  [[ ${#FOUND_ROOT_WORKSPACE_PATHS[@]} -eq 1 && -d "${FOUND_ROOT_WORKSPACE_PATHS[0]}" && ! -f "Package.resolved" ]]; then
```

Comment:
💭  Wondering if it would semantically make more sense to check the presence/absence of `Package.swift` to test if the repo is full-SPM-and-no-xcode-project, instead of testing `Package.resolved` (here + on line 56 + on line 58) 🤔

Reply:
Using `Package.swift` would also make this work for projects that have forgotten to track `Package.resolved`. Granted, that should be a rare occurrence, but it's nice to know it would work in that case, too.

Reply:
Fixed in 72b73ae